traitsui/tests/editors/test_html_editor.py

```diff
@@ -74,32 +255,129 @@ def test_base_url_changed(self):
             model.model_base_url = "/new_dir"
 
     @requires_toolkit([ToolkitName.qt])
-    @mock.patch('webbrowser.open_new')
-    def test_open_externally_qt(self, open_new_mock):
-        from pyface.qt import QtCore, QtWebKit
+    def test_open_internal_link(self):
+        # this test requires Qt because it relies on the link filling up
+        # the entire page through the use of CSS, which isn't supported
+        # by Wx.
```

Comment:
This explains the use of `style='display:block; width: 100%; height: 100%'` in the tests, and the implementation of `qt_mouse_click_web_view` which "Performs a mouse click at the center of the web view".  I'm assuming this was done this way because finding the location of the link itself in the page to be clicked is too difficult.  If it were possible though these tests (or ones similar with possibly different expected behaviors) could be done with wx correct?

Reply:
> I'm assuming this was done this way because finding the location of the link itself in the page to be clicked is too difficult. I'm assuming this was done this way because finding the location of the link itself in the page to be clicked is too difficult.

I believe it is possible. Both Qt and Wx have methods for finding text in the HTML view and having them selected. I think what happened here is that I took the shorter path so we at least have some tests to work with for #1464. The thinking goes: modify/add tests for refactoring, refactor, add failing tests for changing behaviour, change the behaviour, then loop again. It is possible that eventually we will have modified these tests a number of times and we will end up with test support code that more powerful and useful. But until then, we can iterate and improve these test support code while keeping them private here.